manager/kubernetes.go

```diff
@@ -61,7 +61,15 @@ func (m *VolumeManager) PVCreate(name, pvName, fsType, secretNamespace, secretNa
 		pv.Spec.CSI.NodePublishSecretRef = secretRef
 	}
 
-	pv, err = m.ds.CreatePersistentVolume(pv)
+	scheduledCondition := types.GetCondition(v.Status.Conditions, longhorn.VolumeConditionTypeScheduled)
+	if scheduledCondition.Status == longhorn.ConditionStatusFalse {
+		if pv.Annotations == nil {
+			pv.Annotations = map[string]string{}
+		}
+		pv.Annotations[types.PVAnnotationLonghornVolumeSchedulingError] = scheduledCondition.Message
+	}
```

Comment:
I would prefer this should be reconciled in the controller only, so we don't need to have sperate update logic in more than one place. The downside will be some delay for the update, but it will be finally consistent. WDYT?

Reply:
Yes. If the delay is acceptable, I prefer your idea.